src/md.rs

```diff
@@ -96,150 +88,11 @@ pub(crate) fn mess_ratio(
-            flags.insert(MessDetectorCharFlags::ALPHABETIC);
-            if character.is_lowercase() {
-                flags.insert(MessDetectorCharFlags::LOWERCASE);
-                flags.insert(MessDetectorCharFlags::CASE_VARIABLE);
-            } else if character.is_uppercase() {
-                flags.insert(MessDetectorCharFlags::UPPERCASE);
-                flags.insert(MessDetectorCharFlags::CASE_VARIABLE);
-            }
-        } else if !flags.contains(MessDetectorCharFlags::ASCII_GRAPHIC)
-            && !['\x1A', '\u{FEFF}'].contains(&character)
-            && MessDetectorChar::in_category(category, range, &["Cc"], &[], &["Control character"])
-        {
-            flags.insert(MessDetectorCharFlags::UNPRINTABLE);
-        }
-
-        // emoticon
-        if MessDetectorChar::in_category(category, range, &[], &[], &["Emoticons"]) {
-            flags.insert(MessDetectorCharFlags::EMOTICON);
-        }
-
-        // separator
-        if ['｜', '+', '<', '>'].contains(&character)
-            || MessDetectorChar::in_category(category, range, &["Po", "Pd", "Pc"], &["Z"], &[])
-        {
-            flags.insert(MessDetectorCharFlags::SEPARATOR);
-        }
-    }
-
-    // punctuation
-    if MessDetectorChar::in_category(category, range, &[], &["P"], &["Punctuation"]) {
-        flags.insert(MessDetectorCharFlags::PUNCTUATION);
-    }
-
-    // symbol
-    if MessDetectorChar::in_category(category, range, &[], &["N", "S"], &["Forms"]) {
-        flags.insert(MessDetectorCharFlags::SYMBOL);
-    }
-
-    // latin
-    if MessDetectorChar::in_description(name, &["LATIN"]) {
-        flags.insert(MessDetectorCharFlags::LATIN);
-    } else {
-        // cjk
-        if MessDetectorChar::in_description(name, &["CJK"]) {
-            flags.insert(MessDetectorCharFlags::CJK);
-        }
-        // hangul
-        if MessDetectorChar::in_description(name, &["HANGUL"]) {
-            flags.insert(MessDetectorCharFlags::HANGUL);
-        }
-        // katakana
-        if MessDetectorChar::in_description(name, &["KATAKANA"]) {
-            flags.insert(MessDetectorCharFlags::KATAKANA);
-        }
-        // hiragana
-        if MessDetectorChar::in_description(name, &["HIRAGANA"]) {
-            flags.insert(MessDetectorCharFlags::HIRAGANA);
-        }
-        // thai
-        if MessDetectorChar::in_description(name, &["THAI"]) {
-            flags.insert(MessDetectorCharFlags::THAI);
-        }
-    }
-
-    // accentuated
-    if MessDetectorChar::in_description(
-        name,
-        &[
-            "WITH GRAVE",
-            "WITH ACUTE",
-            "WITH CEDILLA",
-            "WITH DIAERESIS",
-            "WITH CIRCUMFLEX",
-            "WITH TILDE",
-        ],
-    ) {
-        flags.insert(MessDetectorCharFlags::ACCENTUATED);
-    }
-
-    // create new object
-    MessDetectorChar {
-        character,
-        flags,
-        unicode_range: range,
-    }
+    detectors.iter().map(|x| x.ratio()).sum()
```

Comment:
why it is here instead of `mean_mess_ratio`?

Reply:
I was trying to redo calculations in case of index ==0 (which I skip now)

i realize it is wrong because it has to be recalculated before the trace as well.


7e085a5 fixes it but ensures double calc most of the time.

There are multiple ways to reduce this double calc including
using `mean_mess_ratio: Option<f32>` or calling a early return function instead of break in the if.

I've tried both and they work, but introduces a bit of complexity so i am keeping it simple for now.

Reply:
Maybe we could use condition `if log_enabled!(Level::Trace) `, see https://docs.rs/env_logger/latest/env_logger/ ?

Reply:
Since there are other places where the trace exists, maybe addressing that could be part of another pr?

Reply:
I don't think so. It is only one place when we have additional calculation because of logs. I could use it to avoid double calculation of mean_mess_ratio.

Reply:
[d442d08](https://github.com/nickspring/charset-normalizer-rs/pull/33/commits/d442d089dc76fbab3ccc8e8261aab75e8156627e)
removes recalc.
I'll look to integrate `if log_enabled!(Level::Trace)`